runtime/src/main/java/org/corfudb/runtime/view/LayoutBuilder.java

```diff
@@ -99,17 +99,17 @@ public LayoutBuilder removeUnresponsiveServer(@Nonnull String endpoint) {
      */
     public LayoutBuilder removeLayoutServer(String endpoint) {
 
-        List<String> layoutServers = layout.getLayoutServers();
-        for (int i = 0; i < layoutServers.size(); i++) {
-            if (layoutServers.get(i).equals(endpoint)) {
-                if (layoutServers.size() == 1) {
-                    throw new LayoutModificationException(
-                            "Attempting to remove all layout servers.");
-                }
-                layoutServers.remove(i);
-                return this;
-            }
+        // Only remove the endpoint if it does not attempt to remove the last remaining layout server
+        if (layout.getLayoutServers().size() == 1 &&
+            layout.getLayoutServers().get(0).equals(endpoint)) {
```

Comment:
perhaps contains() would be more clear here.

Reply:
I'm a bit hesitant on using contains as I feel **equals** is more expressive in this case. We have just **one** server and we check its equality against endpoint. However, let me know if you feel strongly about it and I'll change it as requested.